Subject: [PATCH] time runx too fast after S3
X-BK-ChangeSetKey: shaohua.li@intel.com[torvalds]|ChangeSet|20050105052815|56360
Date:	Wed, 05 Jan 2005 10:34:51 +0000
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.2082, 2005/01/04 21:28:15-08:00, shaohua.li@intel.com

	[PATCH] time runx too fast after S3
	
	After resume from S3, 'date' shows time run too fast.
	
	Signed-off-by: Li Shaohua <shaohua.li@intel.com>
	Signed-off-by: Andrew Morton <akpm@osdl.org>
	Signed-off-by: Linus Torvalds <torvalds@osdl.org>



 time.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)


diff -Nru a/arch/i386/kernel/time.c b/arch/i386/kernel/time.c
--- a/arch/i386/kernel/time.c	2005-01-04 23:23:20 -08:00
+++ b/arch/i386/kernel/time.c	2005-01-04 23:23:20 -08:00
@@ -342,12 +342,13 @@
 		hpet_reenable();
 #endif
 	sec = get_cmos_time() + clock_cmos_diff;
-	sleep_length = get_cmos_time() - sleep_start;
+	sleep_length = (get_cmos_time() - sleep_start) * HZ;
 	write_seqlock_irqsave(&xtime_lock, flags);
 	xtime.tv_sec = sec;
 	xtime.tv_nsec = 0;
 	write_sequnlock_irqrestore(&xtime_lock, flags);
-	jiffies += sleep_length * HZ;
+	jiffies += sleep_length;
+	wall_jiffies += sleep_length;
 	return 0;
 }
 
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

